### PR TITLE
{EventHub} `az eventhubs eventhub create/delete/update/show/list`: Migrating Eventhub Entity cmdlets to AAZ

### PR DESCRIPTION
**Pull request by schaudhari6254888**
**Related command**

**Description**
This PR Adds the `Retention-Description` feature for EventHub Entity. Which include `cleanup-policy`,`retention-time-in-hours`,`tombstone-retention-time-in-hours` parameters. Also Deprecated the `message-retention` parameter , it is replaced with `retention-time-in-hours`

**Testing Guide**

**History Notes**

[EventHub] BREAKING CHANGE: `az eventhubs eventhub create/update`: Remove `--message-retention` parameter, it is replaced by `--retention-time-in-hours`
[EventHub]`az eventhubs eventhub create/update`: Add `--cleanup-policy`, `--retention-time-in-hours` and `--tombstone-retention-time-in-hours` to support `Retention-Description` feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by damodaravadhani** [comment]
Since this PR makes a major version upgrade, external module's tests that used eventhubs cmdlets failed. I was able to run 5 of them successfully. However, the 4 shown below are buggy and fail on a non-eventhub cmdlet in their tests. I upgraded the EventHub api versions manually in the test recordings, but the CI's still complain. @kairu-ms , any suggestions would be appreciated.

image

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/recordings/test_sql_server_ms_support_mgmt.yaml [hunk 1]
```diff
@@ -2810,7 +2810,7 @@ interactions:
       - AZURECLI/2.46.0 (AAZ) azsdk-python-core/1.24.0 Python/3.10.10 (Linux-5.15.0-1033-azure-x86_64-with-glibc2.31)
         VSTS_7b238909-6802-4b65-b90d-184bca47f458_build_220_0
     method: GET
-    uri: https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.EventHub/namespaces/cliehnamespacems02?api-version=2022-01-01-preview
+    uri: https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.EventHub/namespaces/cliehnamespacems02?api-version=2022-10-01-preview
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/recordings/test_sql_server_ms_support_mgmt.yaml:2813]: The namespace commands are still using the old api. You should not replace it here. https://github.com/Azure/azure-cli/blob/d429c2f18abddabc720bcb05d3aedaadd53f0b7d/src/azure-cli/azure/cli/command_modules/eventhubs/aaz/latest/eventhubs/namespace/_create.py#L27-L32

File: src/azure-cli/requirements.py3.windows.txt [hunk 2]
```diff
@@ -44,7 +44,7 @@ azure-mgmt-datamigration==10.0.0
 azure-mgmt-devtestlabs==4.0.0
 azure-mgmt-dns==8.0.0
 azure-mgmt-eventgrid==10.2.0b2
-azure-mgmt-eventhub==10.1.0
```
Comment (kairu-ms) [inline src/azure-cli/requirements.py3.windows.txt:47]: The aaz commands do not rely on python SDK. Please don't change it.

**Review by kairu-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**